src/OrbitGl/MemoryTrack.h

```diff
@@ -8,34 +8,41 @@
 #include <string>
 #include <utility>
 
+#include "AnnotationTrack.h"
+#include "GraphTrack.h"
 #include "Track.h"
-#include "VariableTrack.h"
 #include "Viewport.h"
 
 namespace orbit_gl {
 
-class MemoryTrack final : public VariableTrack {
+constexpr size_t kSystemMemoryTrackSize = 3;
+
+template <size_t Size>
+class MemoryTrack final : public GraphTrack<Size>, public AnnotationTrack {
```

Comment:
Ah, I guess that's why you've been using virtual inheritance? I don't think we should go that route.

There are several ways to avoid virtual inheritance. I assume the simplest one would be to avoid `AnnotationTrack`. It's basically just a function `DrawAnnotation` which can be implemented as a free function.

Another solution would be to make `AnnotationTrack` a template with the base class as a template argument. Then you can inherit from `AnnotationTrack<GraphTrack<Size>>` here.

Reply:
I just looked at `DrawAnnotations`. It doesn't even access anything from `Track`, so it does not need to inherit from `Track`. Hence, it could either be a free function or a class not inherting from `Track`.

Reply:
Thanks for the suggestion.

I notice that only the track size and positions are used in `DrawAnnotations`. So I added 

```
[[nodiscard]] virtual Vec2 GetAnnotatedTrackSize() const = 0; 
[[nodiscard]] virtual Vec2 GetAnnotatedTrackPosition() const = 0;
```
to the `AnnotationTrack`. And `AnnotationTrack` does not inherit from `Track` now.